### PR TITLE
[16.0][ADD] stock_procurement_customer

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

File: stock_procurement_customer/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+
+    _inherit = "stock.picking"
+
+    customer_id = fields.Many2one(
```
Comment (lmignon) [inline stock_procurement_customer/models/stock_picking.py:11]: @rousseldenis This field should be displayed on the picking view if the value is filled and different of the partner_id.

**Review by lmignon — CHANGES_REQUESTED** [review]

File: stock_procurement_customer/models/stock_picking.py (hunk 1 above)
Comment (rousseldenis) [reply stock_procurement_customer/models/stock_picking.py:11]: @lmignon Done

**Review by lmignon — APPROVED** [review]
LGTM (Code + functional)

**Review by adrienpeiffer — APPROVED** [review]
Code review

**Review by lmignon — COMMENTED** [review]
LGTM (Code + functional)

**Review by sebalix — APPROVED** [review]
LG

**Comment by sebalix** [comment]
@rousseldenis do you want to squash your commits before merging?

**Review by jbaudoux — APPROVED** [review]

File: stock_procurement_customer/models/stock_move.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    _inherit = "stock.move"
+
+    def _get_new_picking_values(self):
+        """Prepares a new picking for this move as it could not be assigned to
+        another picking.
+        Add the customer from the procurement group.
+        """
+        res = super()._get_new_picking_values()
+        res["customer_id"] = self.mapped("group_id").customer_id.id
```
Comment (jbaudoux) [inline stock_procurement_customer/models/stock_move.py]:
```suggestion
        res["customer_id"] = self.group_id.customer_id.id
```

**Pull request merged**